src/transformers/models/t5/modeling_t5.py

```diff
@@ -647,7 +647,8 @@ def forward(
     ):
 
         if past_key_value is not None:
-            assert self.is_decoder, "Only decoder can use `past_key_values`"
+            if not self.is_decoder:
+                logger.warning("`past_key_values` is passed to the encoder. Please make sure this is intended.")
```

Comment:
We should raise an `ValueError` here because `past_key_values` is not a valid input for the encoder and should not be passed.

Reply:
Please see the discussion in #15591